### DISCUSSION TITLE
Source term doesn't seem to do anything

### DISCUSSION THREAD

**Discussion by DIsrael-LANL** [usnistgov/fipy#983] [General]
I've got a more complex set of coupled 1-d PDEs, but for now, I've narrowed the error down to this simple example.  Why doesn't the following work?

```
from fipy import Grid1D, CellVariable, TransientTerm, Viewer, ImplicitSourceTerm
from numpy import cos

m = Grid1D(nx=100, Lx=1.)
u = CellVariable(mesh=m, name="u", value=1)
eq = TransientTerm(var=u) == ImplicitSourceTerm(var=u)
vi = Viewer(u)
for t in range(10):
    eq.solve(dt=1.e-2)
    vi.plot()
```

I expect to see exponential growth (this is really a trivially ODE at each point), but instead the solution stays u=1 for all times. 
If I write the source term explicitly, it also does not work.

If I use a source term that is spatially varying, then it works.  (For example, if I set u.setValue=m.x.)

**Comment by wd15** [comment]
It seems like the default coefficient for the ImplicitSourceTerm is `0.0`. I always thought the default value to be `1.0`. Anyway, the following seems to work,

```
from fipy import Grid1D, CellVariable, TransientTerm, Viewer, ImplicitSourceTerm
from numpy import cos

m = Grid1D(nx=100, Lx=1.)
u = CellVariable(mesh=m, name="u", value=1.0)
eq = TransientTerm(var=u) == ImplicitSourceTerm(coeff=1.0, var=u)
vi = Viewer(u)
for t in range(10):
    eq.solve(dt=1.e-2)
    vi.plot()
```
Also, remember to set `value=1.0` in the `CellVariable` to make it a float.

**Reply by guyer** [reply]
Yep, this issue was identified several years ago (#512), but evidently never fixed.
